### ISSUE TITLE
twisted.web.test.test_xmlrpc.XMLRPCClientErrorHandling.test_erroneousResponse can fail depending on PYC contents

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3779]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#3779|
|Type|enhancement|
|Created|2009-04-16 14:47:43Z|
|Branch|https://github.com/twisted/twisted/tree/xmlpyc-3779|

Trying the Twisted trunk test suite on a machine running the Jaunty beta, I got a test failure from `twisted.web.test.test_xmlrpc.XMLRPCClientErrorHandling.test_erroneousResponse`.  It's not intermittent, exactly (I don't think) but for some reason the `xmlrpclib` on my jaunty machine thinks that the .pyc file is a valid-ish xmlrpc response, which causes the request to succeed, which fails the test.

The test could probably use an overhaul (it's trying to do way too much) but if it did just a bit less, it could at least serve up a deterministic pile of invalid data rather than the ever-shifting contents of the .pyc file of the test itself.

<details><summary>Searchable metadata</summary>

```
trac-id__3779 3779
type__enhancement enhancement
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_xmlpyc_3779 branches/xmlpyc-3779
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1239893263000000 1239893263000000
changetime__1239940595000000 1239940595000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

(In [26712]) Branching to 'xmlpyc-3779'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

It's already deterministic.  Python 2.6.2 just thinks that the pyc file is a valid XML-RPC document now.  I filed http://bugs.python.org/issue5767 for this yesterday.

We should probably still pick a simpler pile of garbage to serve up for this test, one that is more likely to trigger a parse error.  However, I'm not sure we can dismiss the current behavior.  Python 2.5.4 and Python 2.6.2 are both going to be parsing complete garbage into `((None,), None)` instead of raising an exception.  This may cause real problems.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

(In [26713]) re [#3779](https://github.com/twisted/twisted/issues/3779) Almost a one-line fix.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph removed owner|
|-|-|

Replying to [exarkun](#note_2):
> It's already deterministic.  Python 2.6.2 just thinks that the pyc file is a valid XML-RPC document now.  I filed http://bugs.python.org/issue5767 for this yesterday.

Yeah, I sort of understood that.  Thanks for the link to the Python issue, I was on my way to do that just now.

> We should probably still pick a simpler pile of garbage to serve up for this test, one that is more likely to trigger a parse error.  However, I'm not sure we can dismiss the current behavior.  Python 2.5.4 and Python 2.6.2 are both going to be parsing complete garbage into `((None,), None)` instead of raising an exception.  This may cause real problems.

Here's the simpler pile of garbage.  Thanks for the link to the Python issue.  You saved me some time filing the bug :).

And thanks for tracking it down to 2.6.2.

I think I'll go file a separate ticket for the bugfix where we address the behavior discovered by this failure...?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @glyph|
|-|-|

Okay, this looks good.  I forced a build on a Jaunty slave, <http://buildbot.twistedmatrix.com/builders/py2.5-select-gc/builds/308>.  Assuming the test passes there, please merge.

I don't think we need a separate ticket for the other bugfix, maybe?  Given that it turns out to be due to having sgmlop installed.  Or maybe we need to tell people not to have sgmlop installed or something?

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set status to `closed`|
|-|-|

(In [26721]) Simplify testing of XML-RPC failure mode to avoid triggering sgmlop parser bug.

This change changes the data expected to be unparseable to a small, static string, rather than the full contents of the PYC of the test case (which, among other things, contains the bytes of a few valid XMLRPC responses).

This allows Twisted's tests to pass on the upcoming Ubuntu release, Jaunty Jackalope, in its default package configuration.  Note however that removing the `python-reportlab-accel` package will fix the older version of this test as well.

Author: glyph

Reviewer: exarkun

Fixes [#3779](https://github.com/twisted/twisted/issues/3779)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
